### PR TITLE
[KEYCLOAK-9361] Consolidate the OpenShift image documentation for 7.3.0 release

### PR DESCRIPTION
**Pull request by iankko**
Various changes to polish the OpenShift image documentation for 7.3 release:

- Replace outdated Openshift Primer example cluster install with links to the latest OCP cluster installation procedure,
- Drop duplicate sections (related to keystores, secrets setup etc.),
- Drop unnecessary sections (related to routes, TLS passthrough mode variables)
- Simplify the 'Initial image deployment' `Getting Started` subsection,
- Consolidate the `Tutorials` DB migration section,
- Move the `Binary Builds` tutorial from `Getting Started` to `Tutorials` section

See relevant git commit messages for further details.

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

### CODE REVIEW

**Comment by iankko** [comment]
@matthewhelmke @stianst PTAL (once got a chance)

Thank you, Jan

**Comment by matthewhelmke** [comment]
I will look at this more closely as soon as I can, but I don't think I will have time to review it all before the 7.3 release tomorrow. However, I should have time to review it before my last day on Friday to try to help get it ready for the first point release later.

File: openshift/topics/advanced_concepts.adoc [hunk 1]
```diff
@@ -227,45 +227,8 @@ $ oc new-app --template={project_templates_version}-https \
...
-and access the {project_name} administrator console at:
-
-* *\https://secure-sso-sso-app-demo.openshift.example.com/auth/admin*
-* *\http://sso-sso-app-demo.openshift.example.com/auth/admin*
-
-using the xref:sso-administrator-setup[administrator account].
-
 [[advanced-concepts-sso-hostname-spi-setup]]
-=== Customizing Hostname for the {project_name} Server
+=== Customizing the Hostname for {project_name} Server
```
Comment (matthewhelmke) [inline openshift/topics/advanced_concepts.adoc:232]: You actually need both instances of "the" in there: Customizing the Hostname for the {project_name} Server.

File: openshift/topics/get_started.adoc [hunk 2]
```diff
@@ -53,658 +53,118 @@ done
...
-----
-.. Add the EAP application secret to the link:https://docs.openshift.com/container-platform/latest/dev_guide/service_accounts.html#default-service-accounts-and-roles[`default`] service account.
-+
-[source,bash,subs="attributes+,macros+"]
-----
-$ oc secrets link default eap-ssl-secret eap-jgroup-secret
-----
 
-===== Deploy Binary Build of the EAP 6.4 / 7.1 JSP Application
+==== Deploying the {project_name} Image using Application Template
```
Comment (matthewhelmke) [inline openshift/topics/get_started.adoc]: "...Using the Application Template" <-needs a "the"

File: openshift/topics/get_started.adoc [hunk 3]
```diff
@@ -53,658 +53,118 @@ done
...
-Replace `--image-stream=jboss-eap71-openshift` parameter with the `--image-stream=jboss-eap64-openshift` one in the following oc command to deploy the JSP application on top of JBoss EAP 6.4 for OpenShift image.
-====
-+
-[source,bash,subs="attributes+,macros+"]
-----
-$ oc new-build --binary=true \
---image-stream=jboss-eap71-openshift \
---name=eap-app
---> Found image 31895a4 (3 months old) in image stream "openshift/jboss-eap71-openshift" under tag "latest" for "jboss-eap71-openshift"
+===== Deploying the Template via OpenShift Web Console
```
Comment (matthewhelmke) [inline openshift/topics/get_started.adoc]: "...via the OpenShift..."

File: openshift/topics/get_started.adoc [hunk 4]
```diff
@@ -53,658 +53,118 @@ done
...
 
-    JBoss EAP 7.1
-    -------------
-    Platform for building and running JavaEE applications on JBoss EAP 7.1
+Alternatively, perform the following steps to deploy the {project_name} template via OpenShift web console:
```
Comment (matthewhelmke) [inline openshift/topics/get_started.adoc:134]: "...via the OpenShift..."

File: openshift/topics/tutorials.adoc [hunk 5]
```diff
@@ -842,6 +797,541 @@ To provide _testuser_ `view` privileges for the _sso-app-demo_, use the OpenShif
...
+Repeat these steps for the `admin` role.
+
+*Create the {project_name} Realm Management User*
+
+. Click *Users* in the *Manage* sidebar to view the user information for the realm.
+. Click *Add User.*
+. Enter a valid *Username* (this example uses the user `appuser`) and click *Save*.
+. Edit the user configuration:
+.. Click the *Credentials* tab in the user space and enter a password for the user (this example uses the password `apppassword`).
+.. Ensure the *Temporary Password* option is set to *Off* so that it does not prompt for a password change later on, and click *Reset Password* to set the user password. A pop-up window prompts for additional confirmation.
```
Comment (matthewhelmke) [inline openshift/topics/tutorials.adoc]: "A pop-up window will prompt you to confirm."

File: openshift/topics/tutorials.adoc [hunk 6]
```diff
@@ -842,6 +797,541 @@ To provide _testuser_ `view` privileges for the _sso-app-demo_, use the OpenShif
...
+[INFO] ------------------------------------------------------------------------
+[INFO] Total time: 3.018 s
+[INFO] Finished at: 2017-06-26T12:22:25+02:00
+[INFO] Final Memory: 35M/310M
+[INFO] ------------------------------------------------------------------------
+----
++
+[IMPORTANT]
+====
+The link:https://github.com/keycloak/keycloak-quickstarts/tree/latest/app-jee-jsp[app-jee-jsp] quickstart requires to configure the adapter, and adapter configuration file (`keycloak.json`) to be present at the `config/` directory in the root of the quickstart to successfully build the quickstart. But since this example configures the adapter later via selected environment variables available for the EAP 6.4 / 7.1 for OpenShift image, it is not necessary to specify the form of `keycloak.json` adapter configuration file at this moment.
```
Comment (matthewhelmke) [inline openshift/topics/tutorials.adoc]: "...quickstart requires you to configure..." "...and that the adapter configuration file...is present in the..."

File: openshift/topics/tutorials.adoc [hunk 7]
```diff
@@ -842,6 +797,541 @@ To provide _testuser_ `view` privileges for the _sso-app-demo_, use the OpenShif
...
+====
+
+[[directory-structure-binary-builds]]
+[start=4]
+. Prepare the directory structure on the local file system.
++
+Application archives in the *deployments/* subdirectory of the main binary build directory are copied directly to the xref:standard-deployments-directory[standard deployments directory] of the image being built on OpenShift. For the application to deploy, the directory hierarchy containing the web application data must be correctly structured.
++
+Create main directory for the binary build on the local file system and *deployments/* subdirectory within it. Copy the previously built WAR archives of both the *service-jee-jaxrs* and *app-jee-jsp* quickstarts to the *deployments/* subdirectory:
```
Comment (matthewhelmke) [inline openshift/topics/tutorials.adoc]: "Create the main directory..."

File: openshift/topics/tutorials.adoc [hunk 8]
```diff
@@ -842,6 +797,541 @@ To provide _testuser_ `view` privileges for the _sso-app-demo_, use the OpenShif
...
+    |__ service.war
+
+1 directory, 2 files
+
+----
++
+[[standard-deployments-directory]]
+[NOTE]
+====
+Location of the standard deployments directory depends on the underlying base image, that was used to deploy the application. See the following table:
```
Comment (matthewhelmke) [inline openshift/topics/tutorials.adoc]: "The location of the standard..."

File: openshift/topics/tutorials.adoc [hunk 9]
```diff
@@ -842,6 +797,541 @@ To provide _testuser_ `view` privileges for the _sso-app-demo_, use the OpenShif
...
+image:images/sso_app_jee_jsp.png[{project_name} Example JSP Application]
+
+Perform the following to test the application:
+
+* Click the *INVOKE PUBLIC* button to access the `public` endpoint that doesn't require authentication.
++
+You should see the *Message: public* output.
+* Click the *LOGIN* button to be redirected for user authentication to the {project_name} server instance against the `demo` realm.
++
+Specify username and password of the {project_name} user configured earlier (`appuser` / `apppassword`). Click *Log in*. The look of the application changes as detailed in the following image:
```
Comment (matthewhelmke) [inline openshift/topics/tutorials.adoc]: "Specify the username..."

**Review by matthewhelmke — CHANGES_REQUESTED** [review]
This is long and I am pleased to request very few changes. Well done!

**Review by matthewhelmke — DISMISSED** [review]
Thank you!

File: openshift/topics/advanced_concepts.adoc (hunk 1 above)
Comment (iankko) [reply openshift/topics/advanced_concepts.adoc:232]: Thanks, Matt! Fixed (see last commit).

File: openshift/topics/get_started.adoc (hunk 3 above)
Comment (iankko) [reply openshift/topics/get_started.adoc]: Fixed.

File: openshift/topics/tutorials.adoc (hunk 5 above)
Comment (iankko) [reply openshift/topics/tutorials.adoc]: Added your suggestion.

File: openshift/topics/tutorials.adoc (hunk 6 above)
Comment (iankko) [reply openshift/topics/tutorials.adoc]: Ditto. Fixed both parts.

File: openshift/topics/tutorials.adoc (hunk 8 above)
Comment (iankko) [reply openshift/topics/tutorials.adoc]: This one too.

File: openshift/topics/tutorials.adoc (hunk 9 above)
Comment (iankko) [reply openshift/topics/tutorials.adoc]: Ditto.

**Review by matthewhelmke — DISMISSED** [review]
LGTM, thank you!

**Comment by iankko** [comment]
Addressed all issues (see last commit) & rebased to current master. Thank you for the suggestions!

**Comment by tanberry** [comment]
Hi all, I'm tagging others in CCS to take a look at where we are with this guide. Jan thank you for all of the months and months that you and Matthew worked on this. 

It sounds like there is still a review needed here and then merge this work, but also a higher-level discussion about what to do with the older xPAS guide. I'll loop in on on the email thread, Jan.

@cwatkins @vikram @ncbaratta

**Comment by hmlnarik** [comment]
@iankko could you please rebase this PR?

**Comment by iankko** [comment]
@hmlnarik Fixed the conflict & rebased against current master. Thanks for heads-up!

**Review by hmlnarik — APPROVED** [review]

**Pull request merged**